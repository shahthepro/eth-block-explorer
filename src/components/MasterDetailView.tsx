import * as React from 'react';
import { MasterDetailWrapper, MasterContainer, DetailsContainer } from './styles/MasterDetailStyles';
import * as PropTypes from 'prop-types';

class MasterDetailView extends React.Component {
  static defaultProps = {
    masterOpen: true
  }

  static propTypes = {
    masterOpen: PropTypes.bool,
    masterSlot: PropTypes.element.isRequired,
    detailSlot: PropTypes.element.isRequired,
  };

  render() {
    const { masterOpen, masterSlot, detailSlot } = this.props as {
      masterOpen: boolean,
      masterSlot: PropTypes.ReactElementLike,
      detailSlot: PropTypes.ReactElementLike,
    };

    return (
      <MasterDetailWrapper>
        <MasterContainer>
          { masterOpen && masterSlot }
        </MasterContainer>
        <DetailsContainer>
          { detailSlot }
        </DetailsContainer>
      </MasterDetailWrapper>
    );
  }
}

export default MasterDetailView;