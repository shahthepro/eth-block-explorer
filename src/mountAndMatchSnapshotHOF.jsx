export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}