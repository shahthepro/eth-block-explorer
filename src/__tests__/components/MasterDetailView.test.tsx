import * as React from 'react';
import * as ReactDOM from 'react-dom';
import MasterDetailView from './../../components/MasterDetailView';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

mountAndMatchSnapshot('MasterDetailView', <MasterDetailView masterSlot={<div>Master</div>} detailSlot={<div>Details</div>} />);
mountAndMatchSnapshot('MasterDetailView masterOpen', <MasterDetailView masterOpen masterSlot={<div>Master</div>} detailSlot={<div>Details</div>} />);
mountAndMatchSnapshot('MasterDetailView autoHeight', <MasterDetailView autoHeight masterSlot={<div>Master</div>} detailSlot={<div>Details</div>} />);
mountAndMatchSnapshot('MasterDetailView autoHeight masterOpen', <MasterDetailView autoHeight masterOpen masterSlot={<div>Master</div>} detailSlot={<div>Details</div>} />);