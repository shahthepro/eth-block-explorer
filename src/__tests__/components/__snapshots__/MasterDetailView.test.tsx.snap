// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <MasterDetailView autoHeight masterOpen> snapshot 1`] = `
<MasterDetailView
  autoHeight={true}
  detailSlot={
    <div>
      Details
    </div>
  }
  masterOpen={true}
  masterSlot={
    <div>
      Master
    </div>
  }
>
  <styled.div
    className="limit-height auto-height"
  >
    <StyledComponent
      className="limit-height auto-height"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "lemnZd",
            "rules": Array [
              "
          height: 100%;
          width: 100%;
          display: flex;
          flex-direction: row;
          background-color: #fff;
          transition: height 0.3s ease;
          &.limit-height {
            max-height: 549px;
          }
          &.auto-height {
            height: auto;
          }
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="limit-height auto-height sc-bdVaJa lemnZd"
      >
        <styled.div
          className=""
        >
          <StyledComponent
            className=""
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "lacFfN",
                  "rules": Array [
                    "
                flex: 280px 0 0;
                transition: flex 0.3s ease, opacity 0.3s ease 0.3s;
                opacity: 1;
                &.master-closed {
                  flex: 0px 0 0;
                  opacity: 0;
                }
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH lacFfN"
            >
              <div>
                Master
              </div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "cXclpg",
                  "rules": Array [
                    "
                flex: auto;
                transition: flex 0.3s ease;
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat cXclpg"
            >
              <div>
                Details
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MasterDetailView>
`;

exports[`matches <MasterDetailView autoHeight> snapshot 1`] = `
<MasterDetailView
  autoHeight={true}
  detailSlot={
    <div>
      Details
    </div>
  }
  masterOpen={true}
  masterSlot={
    <div>
      Master
    </div>
  }
>
  <styled.div
    className="limit-height auto-height"
  >
    <StyledComponent
      className="limit-height auto-height"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "lemnZd",
            "rules": Array [
              "
          height: 100%;
          width: 100%;
          display: flex;
          flex-direction: row;
          background-color: #fff;
          transition: height 0.3s ease;
          &.limit-height {
            max-height: 549px;
          }
          &.auto-height {
            height: auto;
          }
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="limit-height auto-height sc-bdVaJa lemnZd"
      >
        <styled.div
          className=""
        >
          <StyledComponent
            className=""
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "lacFfN",
                  "rules": Array [
                    "
                flex: 280px 0 0;
                transition: flex 0.3s ease, opacity 0.3s ease 0.3s;
                opacity: 1;
                &.master-closed {
                  flex: 0px 0 0;
                  opacity: 0;
                }
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH lacFfN"
            >
              <div>
                Master
              </div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "cXclpg",
                  "rules": Array [
                    "
                flex: auto;
                transition: flex 0.3s ease;
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat cXclpg"
            >
              <div>
                Details
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MasterDetailView>
`;

exports[`matches <MasterDetailView masterOpen> snapshot 1`] = `
<MasterDetailView
  autoHeight={false}
  detailSlot={
    <div>
      Details
    </div>
  }
  masterOpen={true}
  masterSlot={
    <div>
      Master
    </div>
  }
>
  <styled.div
    className="limit-height "
  >
    <StyledComponent
      className="limit-height "
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "lemnZd",
            "rules": Array [
              "
          height: 100%;
          width: 100%;
          display: flex;
          flex-direction: row;
          background-color: #fff;
          transition: height 0.3s ease;
          &.limit-height {
            max-height: 549px;
          }
          &.auto-height {
            height: auto;
          }
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="limit-height  sc-bdVaJa lemnZd"
      >
        <styled.div
          className=""
        >
          <StyledComponent
            className=""
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "lacFfN",
                  "rules": Array [
                    "
                flex: 280px 0 0;
                transition: flex 0.3s ease, opacity 0.3s ease 0.3s;
                opacity: 1;
                &.master-closed {
                  flex: 0px 0 0;
                  opacity: 0;
                }
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH lacFfN"
            >
              <div>
                Master
              </div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "cXclpg",
                  "rules": Array [
                    "
                flex: auto;
                transition: flex 0.3s ease;
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat cXclpg"
            >
              <div>
                Details
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MasterDetailView>
`;

exports[`matches <MasterDetailView> snapshot 1`] = `
<MasterDetailView
  autoHeight={false}
  detailSlot={
    <div>
      Details
    </div>
  }
  masterOpen={true}
  masterSlot={
    <div>
      Master
    </div>
  }
>
  <styled.div
    className="limit-height "
  >
    <StyledComponent
      className="limit-height "
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "lemnZd",
            "rules": Array [
              "
          height: 100%;
          width: 100%;
          display: flex;
          flex-direction: row;
          background-color: #fff;
          transition: height 0.3s ease;
          &.limit-height {
            max-height: 549px;
          }
          &.auto-height {
            height: auto;
          }
        ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="limit-height  sc-bdVaJa lemnZd"
      >
        <styled.div
          className=""
        >
          <StyledComponent
            className=""
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "lacFfN",
                  "rules": Array [
                    "
                flex: 280px 0 0;
                transition: flex 0.3s ease, opacity 0.3s ease 0.3s;
                opacity: 1;
                &.master-closed {
                  flex: 0px 0 0;
                  opacity: 0;
                }
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH lacFfN"
            >
              <div>
                Master
              </div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "cXclpg",
                  "rules": Array [
                    "
                flex: auto;
                transition: flex 0.3s ease;
              ",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat cXclpg"
            >
              <div>
                Details
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</MasterDetailView>
`;
